### PR TITLE
THORN-2061: Switch MP metrics to SmallRye.

### PR DESCRIPTION
**Pull request by michalszynkiewicz**
Motivation
----------
Replace the implementation of MP Metrics with SmallRye metrics.

Modifications
-------------
Metrics code has been moved to SmallRye. Mapping.yml has been replaced
with base-metrics.properties and vendor-metrics.properties

Result
------
Thorntail metrics use SmallRye Metrics now.

- [X] Have you followed the guidelines in our [Contributing](http://wildfly-swarm.io/community/contributing/) document?
- [X] [v2] Have you created a [JIRA](https://issues.jboss.org/browse/THORN) and used it in the commit message?
- [ ] [v4] Have you created a [GitHub Issue](https://github.com/thorntail/thorntail/issues) and used it in the commit message?
- [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/thorntail/thorntail/pulls) for the same issue?
- [X] Have you built the project locally prior to submission with `mvn clean install`? (with mp tck)

-----

### CODE REVIEW

File: build-parent/pom.xml [hunk 1]
```diff
@@ -111,6 +111,9 @@
     <!-- MP Config implementation -->
     <version.wildfly-microprofile-config>1.2.1</version.wildfly-microprofile-config>
 
+    <!-- SmallRye -->
```
Comment (sberyozkin) [inline build-parent/pom.xml]: I'll add a 'hold' label till the smallrye-metrics is released

File: fractions/microprofile/microprofile-metrics/src/main/java/org/wildfly/swarm/microprofile/metrics/MicroprofileMetricsFraction.java [hunk 2]
```diff
@@ -17,11 +17,16 @@
...
 
 import org.wildfly.swarm.spi.api.Fraction;
+import org.wildfly.swarm.spi.api.Module;
 import org.wildfly.swarm.spi.api.annotations.DeploymentModule;
 
 /**
  * @author Heiko W. Rupp
  */
-@DeploymentModule(name = "org.wildfly.swarm.microprofile.metrics", slot = "deployment", export = true, metaInf = DeploymentModule.MetaInfDisposition.IMPORT)
+@DeploymentModule(
```
Comment (sberyozkin) [inline fractions/microprofile/microprofile-metrics/src/main/java/org/wildfly/swarm/microprofile/metrics/MicroprofileMetricsFraction.java:26]: Can you please try and remove it altogether (sorry if it is needed after all :-) )

File: build-parent/pom.xml (hunk 1 above)
Comment (michalszynkiewicz) [reply build-parent/pom.xml]: :+1:

File: fractions/microprofile/microprofile-metrics/src/main/java/org/wildfly/swarm/microprofile/metrics/MicroprofileMetricsFraction.java (hunk 2 above)
Comment (michalszynkiewicz) [reply fractions/microprofile/microprofile-metrics/src/main/java/org/wildfly/swarm/microprofile/metrics/MicroprofileMetricsFraction.java:26]: I see the javadoc for `Fraction` says 

> In the event a logical "fraction" requires no configuration, a subclass of this class is not required.

I will check if it works without this class.

Comment (sberyozkin) [reply fractions/microprofile/microprofile-metrics/src/main/java/org/wildfly/swarm/microprofile/metrics/MicroprofileMetricsFraction.java:26]: Sorry, I only referred to the DeploymentModule annotation

Comment (michalszynkiewicz) [reply fractions/microprofile/microprofile-metrics/src/main/java/org/wildfly/swarm/microprofile/metrics/MicroprofileMetricsFraction.java:26]: Unfortunately, without the fraction or with the fraction but without the `@DeploymentModule` it fails.

Comment (sberyozkin) [reply fractions/microprofile/microprofile-metrics/src/main/java/org/wildfly/swarm/microprofile/metrics/MicroprofileMetricsFraction.java:26]: Thanks Michal, sorry you spent some time on it

File: build-parent/pom.xml [hunk 3]
```diff
@@ -404,6 +404,13 @@
         <version>${version.microprofile-jwt-auth}</version>
       </dependency>
 
+      <!-- SmallRye -->
+
+      <dependency>
+        <groupId>io.smallrye</groupId>
+        <artifactId>smallrye-metrics</artifactId>
+        <version>${version.smallrye.metrics}</version>
```
Comment (kenfinnigan) [inline build-parent/pom.xml:412]: Missing a closing dependency tag, causing the build to fail

Comment (michalszynkiewicz) [reply build-parent/pom.xml:412]: thanks! Seems to be a side effect of updating the branch :| https://github.com/thorntail/thorntail/pull/1030/commits/97f602049cc092d1e64de85aa12a9dfda97218ac#diff-cbe7e4692ea77ee1f632145becb13ed9R413

**Review by sberyozkin — APPROVED** [review]
LGTM, will merge later today to give more time for the comments

**Pull request merged**